### DISCUSSION TITLE
FIXME from the code

### DISCUSSION THREAD

**Discussion by rfreis** [uniswap-python/uniswap-python#344] [General]
Hi. Congratulations for your job done!

Just looking at the repository and as I'm working and understanding the same issue of this line comment 
https://github.com/uniswap-python/uniswap-python/blob/7a908112a64bae831f28fc3e609a7fe6db23a1c4/uniswap/uniswap.py#L336

- I decided to share what I was able to find

```python
q96 = 2**96
def price_to_sqrtp(p):
    """
    Code from https://uniswapv3book.com/docs/milestone_1/calculating-liquidity/
    """
    return int(math.sqrt(p) * q96)
```

**Comment by ErikBjare** [comment]
Could you make a PR with this change? :)

**Comment by Alexz1254** [comment]
It's funny

**Comment by frosik890** [comment]
It´s cool

**Comment by tupik999** [comment]
super

**Comment by marik567** [comment]
super

**Comment by svet345** [comment]
goood))

**Comment by Soulchild07** [comment]
good

**Comment by Rayder07** [comment]
nice

**Comment by ErikBjare** [comment]
Why do the bots keep spamming this thread?!

**Comment by hasirama9091** [comment]
Really good!

**Comment by t270811** [comment]
good

**Comment by Olenazel** [comment]
very good
